pkg/client/oauthclient.go

```diff
@@ -0,0 +1,57 @@
+package client
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/watch"
+
+	oauthapi "github.com/openshift/origin/pkg/oauth/api"
+)
+
+type OAuthClientsInterface interface {
+	OAuthClients() OAuthClientInterface
+}
+
+type OAuthClientInterface interface {
+	Create(obj *oauthapi.OAuthClient) (*oauthapi.OAuthClient, error)
+	List(opts kapi.ListOptions) (*oauthapi.OAuthClientList, error)
+	Get(name string) (*oauthapi.OAuthClient, error)
+	Delete(name string) error
+	Watch(opts kapi.ListOptions) (watch.Interface, error)
```

Comment:
> Do we want to have Watch here, I've grepped through our several clients (Roles, UserIdentityMappings, OAuthAccessTokens) don't implement that. Is it needed here, personally I can't see the use-case for that.

Ah, this is just working through the generic etcd client.  If we get the functionality for free, I'd rather keep this API the same shape as all the others.  It would be exposed via REST anyway, this just gives us parity with curl.